### PR TITLE
Fix get_tx_set() for rectangular tx_sizes > 32x32

### PR DESCRIPTION
**Pull request by ycho**
Also fix uv_tx_type decision for intra mode block.

### CODE REVIEW

**Comment by ycho** [comment]
Wow, look at this.
There is unexpected gain of -0.0373%, which I have no idea where it comes from!!

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.0373 | -0.2968 | -0.2811 |  -0.1200 | -0.0071 | -0.0768 |    -0.1019 |

[AWCY](https://beta.arewecompressedyet.com/?job=master_6237e6_5f%402018-11-15T21%3A04%3A16.438Z&job=fix_get_tx_set_5f%402018-11-15T19%3A42%3A27.611Z)

**Comment by fbossen** [comment]
@ycho You seem to disallow IDTX for 32x32 inter

**Comment by ycho** [comment]
@fbossen You mean the change inside write_tx_blocks()? That function is called for intra mode block only?

**Comment by ycho** [comment]
@fbossen Oh, the first line of change in get_tx_set()? Yes, it seems so.... let me check.

**Comment by fbossen** [comment]
@ycho Yes, the value returned by get_tx_set() is different for 32x32 inter blocks

File: src/context.rs [hunk 1]
```diff
@@ -508,11 +508,11 @@ pub fn has_chroma(
 pub fn get_tx_set(
   tx_size: TxSize, is_inter: bool, use_reduced_set: bool
 ) -> TxSet {
-  let tx_size_sqr_up = tx_size.sqr_up();
   let tx_size_sqr = tx_size.sqr();
-  if tx_size_sqr > TxSize::TX_32X32 {
+
+  if tx_size.width() >= 32 || tx_size.height() >= 32 {
```
Comment (ycho) [inline src/context.rs]: Ooops, I should use && instead of || here, so that any tx_size having 64 as one dim use dct_dct only.

Comment (fbossen) [reply src/context.rs]: I think || is fine. Shouldn't it be > instead of >=?

Comment (ycho) [reply src/context.rs]: Hmmm, I thought wrong again.  It should be:  w > 32 || h > 32. or w >= 64 || h >= 64

Comment (ycho) [reply src/context.rs]: Yes! I was writing that! :) thx-

**Comment by ycho** [comment]
Confirm that fix works!
No change in AWCY for five frames.

|   PSNR | PSNR Cb | PSNR Cr | PSNR HVS |   SSIM | MS SSIM | CIEDE 2000 |
|   ---: |    ---: |    ---: |     ---: |   ---: |    ---: |       ---: |
| 0.0000 |  0.0000 |  0.0000 |   0.0000 | 0.0000 |  0.0000 |     0.0000 |

[AWCY](https://beta.arewecompressedyet.com/?job=master_6237e6_5f%402018-11-15T21%3A04%3A16.438Z&job=fix2_get_tx_set_5f%402018-11-15T21%3A39%3A10.763Z)

[Full sequence AWCY result](https://beta.arewecompressedyet.com/?job=master_6237e6%402018-11-15T21%3A22%3A58.123Z&job=fix2_get_tx_set%402018-11-15T21%3A39%3A23.705Z) will be available soon.

**Pull request merged**